datasetinsights/io/downloader/base.py

```diff
@@ -0,0 +1,45 @@
+import re
+from abc import ABCMeta, abstractmethod
+
+
+class DownloaderRegistry(ABCMeta):
+    """ This class registers DatasetDownloader
+        subclasses based on the PROTOCOL attribute
+
+    """
+
+    registry = {}
+
+    def __new__(cls, name, bases, namespace):
+        protocol = "PROTOCOL"
+        if protocol not in namespace:
```

Comment:
Why is interface validation not part of the `DatasetDownloader ` interface definition? Why is the factory method responsible of performing interface validation? Also, it might be better to raise `NotImplementedError` or other errors that is explicit about interface issues. `RuntimeError` was too generic. 

I think this logic can be simplified if we create an `__init_subclass__` method in `DatasetDownloader ` such as:

```
class DatasetDownloader(metaclass=ABCMeta):

@classmethod
    def __init_subclass__(cls):
        """ Executed during subclass init.

        Raises:
            NotImplementedError: Raised if the subclass does not define
                required variables.
        """
        for var in ["PROTOCAL"]:
            if not hasattr(cls, var):
                raise NotImplementedError(
                    f"Subclass needs to have a constant class variable '{var}'."
                )
```

Reply:
I can use __init__subclasses__ but i don't see the point of metaclass be ABCMeta. From my understanding the point of __init__subclass__ is to avoid having to deal with metaclasses.